src/main/java/com/scottlogic/filter/FilterAuthor.java

```diff
@@ -1,15 +1,18 @@
 package com.scottlogic.filter;
 
-import com.scottlogic.strategy.Filter;
 import com.scottlogic.UserPost;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 public class FilterAuthor implements Filter {
 
     @Override
     public List<UserPost> filter(List<UserPost> inputList, String filterCriteria) {
+        if (inputList == null) {
+            return Arrays.asList();
+        }
```

Comment:
I'll allow this for training but in production code it would seem more sensible to throw an exception on invalid inputs. Consider the responsibility of this class, it's a filter - so it wouldn't make much sense for it to assume what the output should be if it gets a null list. Some handler using this would have the context to decide.

I only started thinking like this recently after discussing with my team lead, this way of thought keeps components reusable.

Reply:
I was curious as to what to return here - I presume I would fill in some criteria given to me in an actual project? or would that be something I'd have to decide?

Reply:
It's the idea that a reusable component won't make assumptions for defaults, so the filter class being given a null won't know to set it to an empty list. Another domain class that uses this in the context would catch the exception and set it to null.

Same output, separating responsibility. Reusable/general classes don't know enough to make assumtpions, whilst domain classes has the context to decide. Does that make sense?

Reply:
Absolutely! Thank you.